pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -46,7 +50,11 @@ type DiffListChange struct {
 }
 
 func Diff(old, new Manifest, opts ...diff.Option) (*diff.Result, error) {
-	return diff.DiffUnstructureds(*old.u, *new.u, opts...)
+	modified, err := normalizeNewSecret(old.u, new.u)
```

Comment:
Only do it if they are Secrets.

Reply:
I changed to check whether secret in `normalizeNewSecret` func